lib/core/util.js

```diff
@@ -219,28 +220,74 @@ function parseKeepAliveTimeout (val) {
   return m ? parseInt(m[1], 10) * 1000 : null
 }
 
-function parseHeaders (headers, obj = {}) {
+/**
+ * @param {string | Buffer} value
+ */
+function headerNameToString (value) {
+  const key = value.toString()
+  return headerNameLowerCasedRecord[key] ?? key.toLowerCase()
+}
+
+/**
+ * @param {string | Buffer} value
+ */
+function headerNameToStringUnsafe (value) {
+  return typeof value === 'string'
+    ? headerNameLowerCasedRecord[value] ?? value.toLowerCase()
+    : tree.lookup(value) ?? value.toString().toLowerCase()
+}
+
+/**
+ * @param {Record<any, any> | (Buffer | string | (Buffer | string)[])[]} headers
+ * @param {boolean} [allowUnsafe]
+ * @param {Record<any, any>} [obj]
+ * @returns
+ */
+function parseHeaders (headers, allowUnsafe, obj) {
```

Comment:
Can you move "allowUnsafe" to the end as to not to break the API

Reply:
Sure.